core-bundle/src/Resources/contao/elements/ContentPlayer.php

```diff
@@ -143,9 +143,12 @@ protected function compile()
 			}
 
 			$objFile = new File($objFileModel->path);
-			$objFile->title = StringUtil::specialchars($strTitle ?: $objFile->name);
 
-			$arrFiles[$objFile->extension] = $objFile;
+			$arrFiles[$objFile->extension] = array
```

Comment:
this was previously an object and now is an array, that would break existing `ce_player` templates. Any reason to not just cast this? Also, this should most likely use the asset helper to generate the path of a file (which would include the asset URL if configured in the root page), not manually append a path.

Reply:
We are talking Contao 5, so breaks are allowed. It does not matter anyway because the `ContentPlayer` element will be replaced with a modern content element fragment before the release of Contao 5.0.

Reply:
> It does not matter anyway because the ContentPlayer element will be replaced with a modern content element fragment before the release of Contao 5.0.

But the old one will still be available and we still need to support that.

Reply:
yeah, which even more so makes it a bad experience to change it for no reason imho

Reply:
It is not changed "for no reason". See https://github.com/contao/contao/pull/4851#discussion_r903085380.

Reply:
I did mean the change is "for no reason" because we can simply cast the array to an object for the template

Reply:
```suggestion
			$arrFiles[$objFile->extension] = (object) array
```
This would be enough to make @aschempp and @fritzmg happy, correct?
If so, I’m in favor ☺️

Reply:
But I‘m against this. There is no point in using a `stdClass` object instead of an array. It is Contao 5 – we are allowed to make backwards incompatible changes. Our BC promise does not even cover templates, so we could even do this in Contao 4.13!

Reply:
yes we are allowed to break. But I dislike to break stuff for no apparent reason just because you touched it 🤷

Reply:
> I dislike to break stuff for no apparent reason

Except in your own PRs 🤣 https://github.com/contao/contao/pull/4839#issuecomment-1158586258

Reply:
BC break reverted in 3832d2462c84cf13325ac3eade58acd8ae8cf453.